alien4cloud/user.go

```diff
@@ -0,0 +1,407 @@
+// Copyright 2019 Bull S.A.S. Atos Technologies - Bull, Rue Jean Jaures, B.P.68, 78340, Les Clayes-sous-Bois, France.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alien4cloud
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// UserService is the interface to the service mamaging users and groups
+type UserService interface {
+	// CreateUser creates a user
+	CreateUser(ctx context.Context, createRequest CreateUserRequest) error
+	// UpdateUser updates a user parameters
+	UpdateUser(ctx context.Context, userName string, updateRequest UpdateUserRequest) error
+	// GetUser returns the parameters of a user whose name is provided in argument
+	GetUser(ctx context.Context, userName string) (*User, error)
```

Comment:
Why returning a pointer here while not using pointers in `GetUsers` and `SearchUsers` slice bellow?

Same remark for groups

Reply:
yes not good, I committed a change to fix this